src/core/agent/services/connectionService.ts

```diff
@@ -66,44 +67,57 @@ class ConnectionService extends AgentService {
     this.props.eventEmitter.on(EventTypes.ConnectionStateChanged, callback);
   }
 
+  onConnectionAdded() {
+    this.props.eventEmitter.on(
+      EventTypes.ConnectionStateChanged,
+      (event: ConnectionStateChangedEvent) => {
```

Comment:
Since this same event is used for pending and confirmed, won't we get an infinite loop? So it resolves, and once the resolution completes, its confirmed and this gets triggered again.

We should check for pending